### ISSUE TITLE
multiprocessing SyncManager method signatures seem inaccurate

### ISSUE THREAD

**Issue by ikelos** [python/typeshed#1778]
The SyncManager in stdlib/3/multiprocessing/managers.pyi has methods such as Value listed below:

```def Value(self, typecode: Any, value: _T) -> _T: ...```

This implies that the following line would provide an int (since the second parameter is an int), however it returns a ValueProxy instead.

```Python 3.6.3 (default, Dec  2 2017, 04:21:25) 
[GCC 7.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import multiprocessing
>>> multiprocessing.Manager().Value("Q", 0)
<ValueProxy object, typeid 'Value' at 0x7fa781db18d0>
```

I'm not entirely sure how to resolve this (since I'm not sure whether ValueProxy is a descendent of Value or not), but I can have a shot at a patch if the fix isn't something quick?  Please just let me know if you need any more information or anything else necessary to help with this...

**Comment by srittau** [comment]
multiprocessing does some pretty heavy trickery, so it's a bit hard to follow what's exactly happening in it. But I agree that `Value()` should return a `ValueProxy`, not `_T`.

**Comment by ikelos** [comment]
Hiya, it's been a year and a half.  Any progress on this?  I'm still getting errors thrown up about a clearly incorrect type?

**Comment by srittau** [comment]
Patches welcome!

**Comment by ikelos** [comment]
I'm not sure I've ever done generics correctly, but hopefully this is close?

[multiprocessing-manager-patch.txt](https://github.com/python/typeshed/files/3339757/multiprocessing-manager-patch.txt)

**Comment by ikelos** [comment]
I forgot to import `Generic`, sorry.  5:S

**Comment by srittau** [comment]
@ikelos Thank you, looks mostly good to me, although the default for `ValueProxy.value` is redundant. Could you send a pull request?
